OpenRA.Mods.Common/HitShapes/Circle.cs

```diff
@@ -48,25 +51,27 @@ public WDist DistanceFromEdge(WVec v)
 
 		public WDist DistanceFromEdge(WPos pos, Actor actor)
 		{
-			var actorPos = actor.CenterPosition;
+			var orientation = actor.Orientation;
+			var actorPos = actor.CenterPosition + LocalOffset.Rotate(orientation);
 
 			if (pos.Z > actorPos.Z + VerticalTopOffset)
-				return DistanceFromEdge(pos - (actorPos + new WVec(0, 0, VerticalTopOffset)));
+				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalTopOffset))).Rotate(-orientation));
```

Comment:
Correcting for the `LocalOffset` here but not in `DistanceFromEdge(WVec v)` smells like a bug.  I would expect to see method calculate the offset relative to center of the actor and then rotate it into the local coordinates, and then `DistanceFromEdge(WVec v)` to account for the `LocalOffset`.